src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/data/CreateActivityRequest.kt

```diff
@@ -0,0 +1,118 @@
+package uk.gov.justice.digital.hmpps.nomisprisonerapi.data
+
+import com.fasterxml.jackson.annotation.JsonInclude
+import io.swagger.v3.oas.annotations.media.Schema
+import java.math.BigDecimal
+import java.time.LocalDate
+import javax.validation.constraints.NotBlank
+import javax.validation.constraints.NotNull
+
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@Schema(description = "Course activity creation request")
+data class CreateActivityRequest(
+
+  @Schema(description = "Code generated from the activity and schedule ids and mapped", required = true)
+  @NotBlank
+  val code: String,
+
+  @Schema(description = "Activity start date", required = true, example = "2022-08-12")
+  @NotNull
+  val startDate: LocalDate,
+
+  @Schema(description = "Activity end date", example = "2022-08-12")
+  val endDate: LocalDate?,
+
+  @Schema(description = "Prison where the activity is to occur", required = true)
+  @NotBlank
+  val prisonId: String,
+
+  @Schema(description = "Room where the activity is to occur (from activity schedule)", required = true)
+  @NotNull
+  val internalLocation: Long,
```

Comment:
interesting, will "they" be providing this or will it be mapped from their own room description?

Reply:
they are providing it !